homeassistant/components/ambient_network/config_flow.py

```diff
@@ -0,0 +1,159 @@
+            self._radius = user_input[CONF_LOCATION][CONF_RADIUS]
+
+            client: OpenAPI = OpenAPI()
+            self._stations = {
+                x[API_STATION_MAC_ADDRESS]: x
+                for x in await client.get_devices_by_location(
+                    self._latitude,
+                    self._longitude,
+                    radius=DistanceConverter.convert(
+                        self._radius,
+                        UnitOfLength.METERS,
+                        UnitOfLength.MILES,
+                    ),
+                )
+            }
+
+            # Filter out indoor stations
+            self._stations = dict(
+                filter(
+                    lambda item: not item[1]
+                    .get(API_STATION_INFO, {})
+                    .get(API_STATION_INDOOR, False),
+                    self._stations.items(),
+                )
+            )
+
+            if self._stations:
+                return await self.async_step_station()
+
+            errors = {"base": "no_stations_found"}
+
+        schema: vol.Schema = self.add_suggested_values_to_schema(
+            vol.Schema(
+                {
+                    vol.Required(
+                        CONF_LOCATION,
+                    ): LocationSelector(LocationSelectorConfig(radius=True)),
+                }
+            ),
+            {
+                CONF_LOCATION: {
+                    CONF_LATITUDE: self.hass.config.latitude,
+                    CONF_LONGITUDE: self.hass.config.longitude,
+                    CONF_RADIUS: CONF_RADIUS_DEFAULT,
+                }
+                if not errors
+                else {
+                    CONF_LATITUDE: self._latitude,
+                    CONF_LONGITUDE: self._longitude,
+                    CONF_RADIUS: self._radius,
+                }
+            },
+        )
+
+        return self.async_show_form(
+            step_id=CONF_USER, data_schema=schema, errors=errors if errors else {}
+        )
+
+    async def async_step_station(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the second step to select the station."""
+
+        if user_input:
+            mac_address = user_input[CONF_STATION]
+            await self.async_set_unique_id(mac_address)
+            self._abort_if_unique_id_configured()
+            return self.async_create_entry(
+                title=get_station_name(self._stations.get(mac_address, {})),
+                data={
+                    CONF_MAC: mac_address,
+                },
+            )
+
+        options: list[SelectOptionDict] = [
+            SelectOptionDict(
+                label=get_station_name(station),
+                value=mac_address,
+            )
+            for mac_address, station in sorted(
+                self._stations.items(),
+                key=lambda item: get_station_name(item[1]),
+            )
```

Comment:
I don't think the sorting here is needed, the select selector has a `sort` setting: https://github.com/home-assistant/core/blob/c78e089526076a6ede9bed02fb67d17182e26d4f/homeassistant/helpers/selector.py#L1099

Reply:
Fixed.